src/moves.rs

```diff
@@ -25,19 +28,62 @@ impl Move {
         Move { dx: -2, dy: 1 },
         Move { dx: -2, dy: -1 },
     ];
+
+    pub fn get_pawn_advance_moves(player: Player, first_move: bool) -> &'static [Move] {
+        match player {
+            Player::White => {
+                if first_move {
+                    &[Move { dx: 0, dy: -1 }, Move { dx: 0, dy: -2 }]
+                } else {
+                    &Self::QUEEN_MOVES[3..4]
+                }
+            }
+            Player::Black => {
+                if first_move {
+                    &[Move { dx: 0, dy: 1 }, Move { dx: 0, dy: 2 }]
+                } else {
+                    &Self::QUEEN_MOVES[2..3]
+                }
+            }
+        }
+    }
+
+    pub fn get_pawn_capture_moves_white() -> &'static [Move] {
```

Comment:
We could change the order of `QUEEN_MOVES` if we really wanted to use a slice from there for these.

Reply:
oh yeah haha damn